eth-riscv-runtime/src/lib.rs

```diff
@@ -116,6 +118,55 @@ pub fn msg_sender() -> Address {
     Address::from_slice(&bytes)
 }
 
+pub fn msg_value() -> U256 {
+    let first: u64;
+    let second: u64;
+    let third: u64;
+    let fourth: u64;
+    unsafe {
+        asm!("ecall", lateout("a0") first, lateout("a1") second, lateout("a2") third, lateout("a3") fourth, in("t0") u64::from(Syscall::CALLVALUE));
+    }
+    U256::from_limbs([first, second, third, fourth])
```

Comment:
We should add an abstraction for U256 conversion stuff, I think -- not sure best way